azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt

```diff
@@ -211,6 +211,8 @@ internal class AudioSessionManager(
         if (initialized) return
         initialized = true
 
+        switchAudioDevice(AudioDeviceSelectionStatus.SPEAKER_REQUESTED)
```

Comment:
Instead of switching the state as a side-effect here, perhaps tell AudioManager to turn on speakerphone before this, so the When() block aligns the state properly.

(context.getSystemService(Context.AUDIO_SERVICE) as AudioManager).speakerPhoneEnabled = true

In onCreate of CallCompositeActivity first init?